### PR TITLE
Import explicit types from `@endo/marshal`

### PR DESCRIPTION
**Pull request by michaelfig**
closes: #XXXX
refs: #XXXX

## Description

Informational for @kriskowal (and @mhofman if interested).  This is the other side of endojs/endo#1025 , which must be merged before this work can continue.

### CODE REVIEW

**Review by mhofman — COMMENTED** [review]
Same nit that I'd prefer if we kept using file extensions in type imports.

I'm wondering if this ends up working without the `maxNodeModulesJsDepth` option because the types are explicitly imported?

**Review by erights — COMMENTED** [review]
If it weren't causing lint errors, it would LGTM ;)

**Comment by mhofman** [comment]
Rebased on top of #4420, added a few more explicit type import that were needed, and included the patch to the endo packages so that one can try this without jumping through hoops.

**Comment by mhofman** [comment]
> If it weren't causing lint errors

It doesn't anymore ;)

**Comment by mhofman** [comment]
Found a few other instances of ambient use, rebased and included the patch for the final version merged on the endo side. I think the best approach is to keep this in draft until @kriskowal bumps the versions, and then just use the first commit here.

File: packages/SwingSet/src/types.js [hunk 1]
```diff
@@ -159,7 +159,7 @@
  * @property {string} 0 Kernel slot designating the device node that is the target of
  * the invocation
  * @property {string} 1 A string naming the method to be invoked
- * @property {CapData} 2 A capdata object containing the arguments to the invocation
+ * @property {import('@endo/marshal').CapData<unknown>} 2 A capdata object containing the arguments to the invocation
```
Comment (mhofman) [inline packages/SwingSet/src/types.js:162]: @warner is this the right type?

**Pull request closed without merging**